### ISSUE TITLE
Editor crashes upon creating a new animation in the animation player node.

### ISSUE THREAD

**Issue by NoGodDamnName** [godotengine/godot#8938]
I've decided to go on ahead and report my first potential bug: that the animation player is faulty. 
Upon creating a new animation with the animation player node, the editor simply crashes. I don't think i need to write the steps to reproduce, just try to create a new animation.

Also the textbox to name your new animation is overlapping with dialog text. Clicking OK to create your animation will then crash everything.

My apologies if this issue was already reported i don't know how to use github's search bar apparently.

**Comment by mcanders** [comment]
This isn't caused by creating new animations.  For me, godot segfaults whenever i mouse over the animation timeline (on new or pre-existing animations).  The segfault is in animation_editor.cpp line 2674 because "mb" is null:
```cpp
else if (mb->get_button_mask() & BUTTON_MASK_MIDDLE) {
	int rel = mm->get_relative().x;
	float relf = rel / _get_zoom_scale();
	h_scroll->set_value(h_scroll->get_value() - relf);
}
```
mb (of type Ref<InputEventKey>) is referenced within the Ref<InputEventMouseMotion> (mm) block.  When an event of type "InputEventMouseMotion" is encountered, it attempts to reference the variable for an event of type "InputEventKey".  I fixed this by swapping out the mb references for the correct mm references.

This was caused by @reduz's recent InputEvent changes.  He did warn us:

> this might cause bugs I haven't found yet..

I am throwing together a pr now.
